### PR TITLE
Add a status request-response for editors

### PR DESCRIPTION
**Pull request by aozgaa**
The response gives the tsserver version, needed for telemetry.

### CODE REVIEW

**Comment by sheetalkamat** [comment]
May be would be a good idea to add test for this api in tsserverProjectSystem that checks on this command session returns ts.version

File: src/server/protocol.ts [hunk 1]
```diff
@@ -216,6 +217,27 @@ namespace ts.server.protocol {
         projectFileName?: string;
     }
 
+    /**
```
Comment (andy-ms) [inline src/server/protocol.ts]: All of the *Request interfaces "request xxx information from the server" so I would just omit the comment.

File: src/server/protocol.ts [hunk 2]
```diff
@@ -216,6 +217,27 @@ namespace ts.server.protocol {
...
+     * Requests status information from the server
+     */
+    export interface StatusRequest extends Request {
+        command: CommandTypes.Status;
+    }
+
+    export interface StatusResponseBody {
+        /**
+         * the version of the currently-running server.
```
Comment (andy-ms) [inline src/server/protocol.ts]: Nit: We don't version the server independently of TypeScript as a whole (like we do with services) -- so I would say "The TypeScript version (`ts.version`)"

File: src/server/protocol.ts (hunk 1 above)
Comment (aozgaa) [reply src/server/protocol.ts]: removed.

File: src/server/protocol.ts (hunk 2 above)
Comment (aozgaa) [reply src/server/protocol.ts]: changed.

**Comment by aozgaa** [comment]
@sheetalkamat I didn't find a reasonable spot to put the test in `tsserverProjectSystem.ts`, so I placed it in `session.ts` instead. Did I miss a section of the former file where we check various requests? Is the latter okay?

**Pull request merged**